### PR TITLE
Instrument `RecentBlocksCache` with metrics

### LINKED ISSUE #4391: Metrics to track chain head caching
We want some Prometheus metrics to accurately track hit rates and other stats from #4154.

### PR DESCRIPTION
**Pull request by neysofu**
Fixes #4391. These new metrics could help us with identifying and root-causing performance and load issues on the database.

### CODE REVIEW

File: store/postgres/src/chain_store.rs [hunk 1]
```diff
@@ -1877,6 +1954,42 @@ mod recent_blocks_cache {
...
+                .chain_head_cache_oldest_block_num
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .set(self.earliest_block().map(|b| b.ptr.number).unwrap_or(0) as f64);
+
+            self.metrics
+                .chain_head_cache_latest_block_num
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .set(self.earliest_block().map(|b| b.ptr.number).unwrap_or(0) as f64);
```
Comment (leoyvens) [inline store/postgres/src/chain_store.rs:1990]: This is using `earliest_block()`, shouldn't it be using `chain_head()`?

File: store/postgres/src/chain_store.rs [hunk 2]
```diff
@@ -1877,6 +1954,42 @@ mod recent_blocks_cache {
...
+                .inc();
+        }
+
+        fn record_cache_miss(&self) {
+            self.metrics
+                .chain_head_cache_misses
+                .get_metric_with_label_values(&[&self.network])
+                .unwrap()
+                .inc();
+        }
```
Comment (leoyvens) [inline store/postgres/src/chain_store.rs:1971]: You could move `fn record_cache_hit` and `fn record_cache_miss` into `ChainStoreMetrics`.

**Review by leoyvens — APPROVED** [review]

File: store/postgres/src/chain_store.rs (hunk 1 above)
Comment (neysofu) [reply store/postgres/src/chain_store.rs:1990]: Good catch!

File: store/postgres/src/chain_store.rs (hunk 2 above)
Comment (neysofu) [reply store/postgres/src/chain_store.rs]: :+1: fixed in https://github.com/graphprotocol/graph-node/pull/4440/commits/efbf522284bd3cc705939d4419ed7343d77e9564.

**Pull request merged**